src/drivers/uavcan/actuators/esc.cpp

```diff
@@ -230,3 +230,19 @@ void UavcanEscController::orb_pub_timer_cb(const uavcan::TimerEvent &)
 		_esc_status_pub = orb_advertise(ORB_ID(esc_status), &_esc_status);
 	}
 }
+
+uint8_t UavcanEscController::check_escs_status()
+{
+	int esc_status_flags = 0;
+
+
+	for (int index = 0; index < esc_status_s::CONNECTED_ESC_MAX; index++) {
+
+		if (hrt_elapsed_time(&_esc_status.esc[index].timestamp) < 800000.0f && _esc_status.esc[index].timestamp > 0) {
```

Comment:
if you reverse the order of the two statements you can avoid the (presumably more expensive) `hrt_elapsed_time()` call when the timestamp is invalid.